idTip.lua

```diff
@@ -339,37 +339,62 @@ local f = CreateFrame("frame")
 f:RegisterEvent("ADDON_LOADED")
 f:SetScript("OnEvent", function(_, _, what)
   if what == "Blizzard_AchievementUI" then
-    if AchievementFrameAchievementsContainer then
-      for i,button in ipairs(AchievementFrameAchievementsContainer.buttons) do
-        hookScript(button, "OnEnter", function()
-          GameTooltip:SetOwner(button, "ANCHOR_NONE")
+    -- common functions between branches
+    local achievement_enter = function(button)
+      GameTooltip:SetOwner(button, "ANCHOR_NONE")
+      GameTooltip:SetPoint("TOPLEFT", button, "TOPRIGHT", 0, 0)
+      addLine(GameTooltip, button.id, kinds.achievement)
+      GameTooltip:Show()
+    end
```

Comment:
Can we these local functions out to top-level scope? If the functions are pure and don't need the local closure, I strongly prefer it.